pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -672,18 +678,14 @@ func (s *DiscoveryServer) pushConnection(con *XdsConnection, pushEv *XdsEvent) e
 		return nil
 	}
 
-	if err := con.modelNode.SetServiceInstances(pushEv.push.Env); err != nil {
-		return err
-	}
-	if util.IsLocalityEmpty(con.modelNode.Locality) {
-		// Get the locality from the proxy's service instances.
-		// We expect all instances to have the same locality. So its enough to look at the first instance
-		if len(con.modelNode.ServiceInstances) > 0 {
-			con.modelNode.Locality = util.ConvertLocality(con.modelNode.ServiceInstances[0].GetLocality())
-		}
```

Comment:
For sidecar type, this still has some effect.

Reply:
done